### PR TITLE
feat: Resource browser cluster interactions change

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes will include,
- New cluster change interactions.
- Fix & interaction updates for Cluster not reachable handling.
- Fix for browser back button is not working properly.
- Showing header in NotAuthorised state
- Implemented abort controller

Fixes - [AB#1765](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/1765)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by visiting Resource Browser page & performing different actions.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/services/service.types.ts [hunk 1]
```diff
@@ -181,9 +181,10 @@ export interface ClusterListResponse extends ResponseType {
 }
 
 export interface Cluster {
-    id: number,
-    cluster_name : string,
+    id: number
+    cluster_name: string
```
Comment (vivek-devtron) [inline src/services/service.types.ts:185]: We should follow camelCase here

File: src/components/ResourceBrowser/ResourceList/ResourceList.component.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import { ReactComponent as ClusterIcon } from '../../../assets/icons/ic-cluster.svg'
+import { ReactComponent as NamespaceIcon } from '../../../assets/icons/ic-env.svg'
+import { ReactComponent as ErrorIcon } from '../../../assets/icons/ic-error-exclamation.svg'
+import { getCustomOptionSelectionStyle } from '../../v2/common/ReactSelect.utils'
+import { NAMESPACE_NOT_APPLICABLE_TEXT } from '../Constants'
+
+export const clusterUnreachableTippyContent = (errorMsg: string) => {
+    return (
+        <div>
+            <span className="fs-12 fw-6 lh-18">Cluster is not reachable</span>
```
Comment (vivek-devtron) [inline src/components/ResourceBrowser/ResourceList/ResourceList.component.tsx:13]: Pleae move the messaging to const

**Review by vivek-devtron — APPROVED** [review]

File: src/services/service.types.ts (hunk 1 above)
Comment (sohel-rp) [reply src/services/service.types.ts:185]: It's coming from backend

File: src/components/ResourceBrowser/ResourceList/ResourceFilterOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            handleFilterChanges(event.target.value, resourceList)
+            setSearchApplied(true)
+        }
+    }
+
+    const handleOnChangeSearchText = (event): void => {
+        setSearchText(event.target.value)
+    }
+
+    const handleClusterChange = (selected): void => {
```
[the remark this answers is not in the document]
Comment (sohel-rp) [reply src/components/ResourceBrowser/ResourceList/ResourceFilterOptions.tsx]: Done

File: src/components/ResourceBrowser/ResourceList/ResourceList.component.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/ResourceBrowser/ResourceList/ResourceList.component.tsx]: Done

**Pull request merged**